Assets/Scripts/Models/Time/TimeManager.cs

```diff
@@ -130,6 +134,16 @@ public void Update(float time)
         // Systems that update every frame.
         InvokeEvent(EveryFrame, time);
 
+        if (nextFrameActions.Count > 0)
+        {
+            for (int i = 0; i < nextFrameActions.Count; i++)
```

Comment:
Being newer to C# than Java, I prefer having the same keyword personally. And I used to use foreach at every opportunity, not fully believing that it could be *that* bad (and being lazy)... but then with my recent attempts at optimizing I've been trying everything I can think of, and seeing what has an effect... and was astounded at the difference changing a few of the larger foreach loops to for loops had in garbage collection